docs/concepts/resource_flavor.md

```diff
@@ -0,0 +1,87 @@
+# ResourceFlavor object
+
+Resources in a cluster are typically not homogeneous. Resources could differ in:
+- pricing and availability (ex: spot vs on-demand VMs)
+- architecture (ex: x86 vs ARM CPUs)
+- brands and models (ex: Radeon 7000 vs Nvidia A100 vs T4 GPUs)
+
+A ResourceFlavor is an object that represents these resource variations and
+allows you to associate them with node labels and taints.
+
+**Note**: If your cluster is homogeneous, you can use an [empty ResourceFlavor](#empty-resourceflavor)
+instead of adding labels to custom ResourceFlavors.
+
+A sample ResourceFlavor looks like the following:
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1alpha2
+kind: ResourceFlavor
+metadata:
+  name: spot
+nodeSelector:
+  instance-type: spot
+taints:
+- effect: NoSchedule
+  key: spot
+  value: "true"
+```
+
+You can use the `.metadata.name` to reference a ResourceFlavor from a
+ClusterQueue in the `.spec.resources[*].flavors[*].name` field.
+
+## ResourceFlavor labels
+
+**Requires Kubernetes 1.23 or newer**
+
+To associate a ResourceFlavor with a subset of nodes of your cluster, you can
+configure the `.nodeSelector` field with matching node labels that uniquely identify
+the nodes. If you are using [cluster autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler)
+(or equivalent controllers), make sure it is configured to add those labels when
+adding new nodes.
+
+To guarantee that the workload Pods run on the nodes associated to the flavor
+that Kueue decided that the workload should use, Kueue performs the following
+steps:
+
+1. When admitting a workload, Kueue evaluates the
+   [`.nodeSelector`](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector)
+   and [`.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution`](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity)
+   fields in the PodSpecs of your [Workload](workload.md) against the
+   ResourceFlavor labels.
```

Comment:
Add:

ResourceFlavors that don't match the node affinity of the Workload cannot be assigned to a Workload's podSet.

We should probably add examples about this, but it could be a follow up.